File: backend/TicketTracker.Api/Entities/Ticket.cs
using TicketTracker.Api.Entities.Enums;

namespace TicketTracker.Api.Entities;

public class Ticket
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public TicketType Type { get; set; }
    public TicketState State { get; set; }

    public Guid TeamId { get; set; }
    public Team Team { get; set; } = null!;

    public Guid? EpicId { get; set; }
    public Epic? Epic { get; set; }

    public Guid CreatedByUserId { get; set; }
    public User CreatedByUser { get; set; } = null!;

    public Guid? AssigneeUserId { get; set; }
    public User? AssigneeUser { get; set; }

    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
}

File: frontend/src/types/ticket.ts
export type TicketType = 'bug' | 'feature' | 'fix'

export type TicketState =
  | 'new'
  | 'ready_for_implementation'
  | 'in_progress'
  | 'ready_for_acceptance'
  | 'done'

export const TICKET_TYPES: TicketType[] = ['bug', 'feature', 'fix']

export const TICKET_STATES: TicketState[] = [
  'new',
  'ready_for_implementation',
  'in_progress',
  'ready_for_acceptance',
  'done',
]

export const TICKET_TYPE_LABELS: Record<TicketType, string> = {
  bug: 'Bug',
  feature: 'Feature',
  fix: 'Fix',
}

export const TICKET_STATE_LABELS: Record<TicketState, string> = {
  new: 'New',
  ready_for_implementation: 'Ready for implementation',
  in_progress: 'In progress',
  ready_for_acceptance: 'Ready for acceptance',
  done: 'Done',
}

export interface Ticket {
  id: string
  teamId: string
  epicId: string | null
  title: string
  body: string
  type: TicketType
  state: TicketState
  createdByUserId: string
  assigneeUserId: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateTicketRequest {
  teamId: string
  epicId?: string | null
  title: string
  body: string
  type: TicketType
}

export interface UpdateTicketRequest {
  teamId: string
  epicId?: string | null
  title: string
  body: string
  type: TicketType
  state: TicketState
}

export interface TicketListFilters {
  teamId?: string
  type?: TicketType
  epicId?: string
  state?: TicketState
  titleSearch?: string
}
